src/Controller.php

```diff
@@ -609,11 +612,21 @@ public static function wp2staticProcessQueue() : void {
                         WsLog::l( 'No deployment add-ons are enabled, skipping deployment.' );
                     } else {
                         WsLog::l( 'Starting deployment' );
-                        do_action(
-                            'wp2static_deploy',
-                            ProcessedSite::getPath(),
-                            $deployer
-                        );
+                        $query = "SELECT GET_LOCK('wp2static_jobs_deploying', 30) AS lck";
+                        $locked = intval( $wpdb->get_row( $query )->lck );
+                        if ( ! $locked ) {
+                            WsLog::l( 'Failed to acquire "wp2static_jobs_deploying" lock.' );
+                            return;
+                        }
+                        try {
+                            do_action(
+                                'wp2static_deploy',
+                                ProcessedSite::getPath(),
+                                $deployer
+                            );
+                        } finally {
+                            $wpdb->query( "DO RELEASE_LOCK('wp2static_jobs_deploying')" );
```

Comment:
If this ever fails, do we have a way to manually clean up, like with caches?

Reply:
In testing, releasing here was not even necessary because WP apparently closes the connection after the request, which implicitly releases the locks. If we ever run into a problem with a customized WP or something, to turn this off.